test/net_http_test.rb

```diff
@@ -0,0 +1,389 @@
+require 'test_helper'
+require 'semian/net_http'
+require 'thin'
+
+class TestNetHTTP < MiniTest::Unit::TestCase
+  class RackServer
+    def self.call(env)
+      response_code = env['REQUEST_URI'].delete("/")
+      response_code = '200' if response_code == ""
+      [response_code, {'Content-Type' => 'text/html'}, ['Success']]
+    end
+  end
+
+  PORT = 31_050
+  TOXIC_PORT = PORT + 1
+
+  DEFAULT_SEMIAN_OPTIONS = {
+    tickets: 3,
+    success_threshold: 1,
+    error_threshold: 3,
+    error_timeout: 10,
+  }
+
+  def test_semian_identifier
+    with_server do
+      Net::HTTP.start("localhost", TOXIC_PORT) do |http|
+        assert_equal "http_localhost_#{TOXIC_PORT}", http.semian_identifier
+      end
+    end
+  end
+
+  def test_trigger_open
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        open_circuit!
```

Comment:
I think this is a good example of a test helper that shouldn't have a default argument, because it makes the tests more confusing to read. I'd define the URL you're testing at the start of the test and then pass it in here.